gtsam/hybrid/tests/testDCSAM.cpp

```diff
@@ -0,0 +1,685 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    testDCSAM.cpp
+ * @brief   Tests for the DCSAM optimization algorithm
+ * @author  Kevin Doherty
+ * @author  Varun Agrawal
+ * @date    March 2025
+ */
+
+#include <gtsam/discrete/DecisionTreeFactor.h>
+#include <gtsam/discrete/DiscreteMarginals.h>
+#include <gtsam/geometry/Pose2.h>
+#include <gtsam/hybrid/DCSAM.h>
+#include <gtsam/hybrid/DiscreteBoundaryFactor.h>
+#include <gtsam/hybrid/HybridGaussianFactor.h>
+#include <gtsam/hybrid/HybridNonlinearFactor.h>
+#include <gtsam/hybrid/HybridNonlinearFactorGraph.h>
+#include <gtsam/inference/BayesNet-inst.h>
+#include <gtsam/nonlinear/GaussNewtonOptimizer.h>
+#include <gtsam/nonlinear/ISAM2.h>
+#include <gtsam/nonlinear/Symbol.h>
+#include <gtsam/sam/BearingRangeFactor.h>
+#include <gtsam/slam/BetweenFactor.h>
+#include <gtsam/slam/PriorFactor.h>
+
+// Include for test suite
+#include <CppUnitLite/TestHarness.h>
+
+using namespace gtsam;
+
+using symbol_shorthand::C;
+using symbol_shorthand::D;
+using symbol_shorthand::L;
+using symbol_shorthand::X;
+
+const double tol = 1e-7;
+
+/******************************************************************************/
+
+/*
+ * Test simple DiscretePriorFactor. We construct a DiscreteFactor factor
+ * graph with a single binary variable d1 and a single custom
+ * DiscretePriorFactor p(d1) specifying the distribution:
+ * p(d1 = 0) = 0.1, p(d1 = 1) = 0.9.
+ *
+ * After constructing the factor graph, we validate that the marginals are
+ * correct (i.e. they match the input p(d1)) and the most probable estimate
+ * (computed by maximizing p(d1) over possible values of d1) is 1.
+ */
+TEST(DCSAM, DiscretePriorFactor) {
```

Comment:
Strictly speaking this isn't a `DCSAM` test, it is more like an example of using the GTSAM discrete factor graph machinery (which was useful to me at the time it was originally written, because we [in John's lab] were still learning how this stuff worked). It probably could be removed if there are other discrete examples / tests (or moved to be co-located with those if you guys think it is useful).